src/java_bytecode/java_string_library_preprocess.cpp

```diff
@@ -285,32 +285,27 @@ exprt::operandst java_string_library_preprocesst::process_parameters(
   return process_operands(ops, loc, symbol_table, init_code);
 }
 
-/// Creates a string_exprt from the input exprt and adds it to processed_ops
-/// \param processed_ops: the list of processed operands to populate
-/// \param op_to_process: a list of expressions
+/// Creates a string_exprt from the input exprt
+/// \param op_to_process: an operand of a type which implements char sequence
 /// \param loc: location in the source
 /// \param symbol_table: symbol table
 /// \param init_code: code block, in which declaration of some arguments may be
 ///   added
-void java_string_library_preprocesst::process_single_operand(
-  exprt::operandst &processed_ops,
+/// \return the processed operand
+exprt java_string_library_preprocesst::process_single_operand(
```

Comment:
While we're at it, could we rename that function?
Suggestion: `convert_exprt_to_string_exprt`